src/serializer/ResidualHeapSerializer.js

```diff
@@ -324,11 +324,20 @@ export class ResidualHeapSerializer {
       );
     }
 
+    // TODO #2259: Make deduplication in the face of leaking work for custom accessors
+    let isCertainlyLeaked = !obj.mightNotBeHavocedObject();
```

Comment:
Do you see a away to consolidate these two instances of iterating over properties?

If not, it's fine as it is.

Reply:
I will look into it, and submit a separate pull request if I come up with something.